docs/parameters.rst

```diff
@@ -2,58 +2,256 @@
+.. code:: python
 
-- fill_missing(``dict``): Dictionary of format {"method": [col_list]} to indicate the method (``mean``/``median``) to be applied on specified col_list. For example: {"mean":["col a"],"median":["col b]}
+    dtype: str
+    example: "/Users/home/datasets/titanic.csv"
 
-- fill_values(``dict``): Dictionary with keys as column names and values that fill the null records in corresponding column.
+- **test_df_path**
 
-- one_hot(``bool``): ``True`` if one hot encoding desired. Default = ``False``.
+Path to the test dataset. Path should point to a file of one of the supported extensions.
+For list of allowed extensions see :py:mod:`preprocessy.input`.
 
-- remove_outliers(``bool``): ``True`` if outlier records to be removed. If ``False`` then ``replace`` parameter must be given. Default = ``True``
+.. code:: python
 
-- replace(``bool``): Boolean value to indicate if the outliers need to be replaced by ``-999``. Default = False.
+    dtype: str
+    example: "/Users/home/datasets/titanic_test.csv"
 
-- first_quartile(``float``): Float value less than 1 and represents the first percentile marker. For more see :py:mod:`preprocessy.outliers`
+- **target_label**
 
-- third_quartile(``float``): Float value less than 1 and represents the other percentile marker. For more see :py:mod:`preprocessy.outliers`
+Name of the target column.
 
-- type(``str``): The type of Scaler to be used. Options include "MinMaxScaler" | "BinaryScaler" | "StandardScaler".
+.. code:: python
 
-- columns(``list``): List of columns in the dataframe.
+    dtype: str
+    example: "Survived"
 
-- is_combined(``bool``): Parameter to determine whether columns should be scaled together as a group
+- **cat_cols**
 
-- threshold(``dict``): Dictionary of threshold values where the key is the column name and the value is the threshold for that column.
+List of column names provided by user indicating these columns are to be encoded categorically. If ``None`` then ``Preprocessy`` calculates and identifies the columns on its own.
 
-- score_func(``callable``): Function taking two arrays X and y, and returning a pair of arrays
-                     ``(scores, pvalues)`` or a single array with scores. ``score_func`` can be custom
-                     or used from ``sklearn.feature_selection``
+.. code:: python
 
-- k(``int``): Number of top features to select.
+    dtype: list[str]
+    example: ["Gender", "Cabin", "Embarked"]
 
-- test_size(``float,int``): Size of test set after splitting. Can take values from 0 - 1 for float point values, 0 - Number of samples for integer values. Is complementary to train size.
+- **ord_dict**
 
-- train_size(``float,int``): Size of train set after splitting. Can take values from 0 - 1 for float point values, 0 - Number of samples for integer values. Is complementary to test size.
+Ordinal attributes require an associated weight mapping. This parameter is mapping from attribute column
+name to weight mapping.
```

Comment:
Additionally we can write: Weight mapping is a dict specifying the weight associated with each unique value of the attribute in consideration.